cpp/src/decisiontree/kernels/quantile.cuh

```diff
@@ -39,44 +39,66 @@ __global__ void get_all_quantiles(const T* __restrict__ data, T* quantile, const
 	return;
 }
 
-template<typename T>
-void preprocess_quantile(const T* data, const unsigned int* rowids, const int n_sampled_rows, const int ncols, const int rowoffset, const int nbins, std::shared_ptr<TemporaryMemory<T>> tempmem) {
+template<typename T, typename L>
+void preprocess_quantile(const T* data, const unsigned int* rowids, const int n_sampled_rows, const int ncols, const int rowoffset, const int nbins, std::shared_ptr<TemporaryMemory<T, L>> tempmem) {
+
+	// Dynamically determine batch_cols (number of columns processed per loop iteration) from the available device memory.
+	size_t free_mem, total_mem;
+	CUDA_CHECK(cudaMemGetInfo(&free_mem, &total_mem));
```

Comment:
There's some amount of uncertainty about usage of mem-get-info API for subsequent allocations as in PR #558. So, until #558 is concluded properly, is there a way for you to statically do this, without using this API?

Reply:
We can leave this commented out for now, and conservatively process only a column at a time (i.e., hardcode `batch_cols` to 1) until #558 is concluded, and an appropriate `getMemInfo` function is provided for the `deviceAllocator` interface.

Statically computing this would have its caveats as we'd need to make various assumptions about what is running on the GPU at a given time.

Reply:
Remove cudaMemGetInfo and replaced with a workaround until equivalent functionality is available